src/Configuration/Config.Json/src/JsonConfigurationFileParser.cs

```diff
@@ -19,89 +17,70 @@ internal class JsonConfigurationFileParser
         private readonly Stack<string> _context = new Stack<string>();
         private string _currentPath;
 
-        private JsonTextReader _reader;
-
         public static IDictionary<string, string> Parse(Stream input)
             => new JsonConfigurationFileParser().ParseStream(input);
 
         private IDictionary<string, string> ParseStream(Stream input)
         {
             _data.Clear();
-            _reader = new JsonTextReader(new StreamReader(input));
-            _reader.DateParseHandling = DateParseHandling.None;
-
-            var jsonConfig = JObject.Load(_reader);
 
-            VisitJObject(jsonConfig);
+            using (var reader = new StreamReader(input))
+            using (JsonDocument doc = JsonDocument.Parse(reader.ReadToEnd(),
+                new JsonReaderOptions { CommentHandling = JsonCommentHandling.Skip }))
+            {
+                VisitElement(doc.RootElement);
+            }
 
             return _data;
         }
 
-        private void VisitJObject(JObject jObject)
-        {
-            foreach (var property in jObject.Properties())
+        private void VisitElement(JsonElement element) {
+            foreach (var property in element.EnumerateObject())
             {
                 EnterContext(property.Name);
-                VisitProperty(property);
+                VisitValue(property.Value);
                 ExitContext();
             }
         }
 
-        private void VisitProperty(JProperty property)
+        private void VisitValue(JsonElement value)
         {
-            VisitToken(property.Value);
-        }
-
-        private void VisitToken(JToken token)
-        {
-            switch (token.Type)
-            {
-                case JTokenType.Object:
-                    VisitJObject(token.Value<JObject>());
+            switch (value.Type) {
+                case JsonValueType.Object:
+                    VisitElement(value);
```

Comment:
Are we generally OK with recursive calls like this?

Reply:
The flow of the code isn't really changed from before which was recursive, config should only be loaded once on startup for an app typically, or whenever the file changes, which should be pretty infrequent.